src/Filters/atoms/DateTimeBlock.tsx

```diff
@@ -125,16 +125,27 @@ const DateTimeBlock : React.FC<IProps> = ({
   const [displayMinutes, setDisplayMinutes] = useState<string>(getFormattedTime(convertMinutes));
 
   const setDateHours = useCallback(({target: {value}}: React.ChangeEvent<HTMLInputElement>) => {
-    const hourRegex  = /^[0-1]{0,1}[0-9]{0,1}$|(^2[0-4])$/;
-    if (!hourRegex.test(value)) {
-      return;
+    let newVal = '0';
+    if(value.length === 1){
+      if(value === '0'){
+        newVal = ('0');
+      } else {
+        newVal = ('0' + value).slice(-2) ;
+      }
+    } else {
+      newVal = value.slice(-2);
+    }
+    if(Number(newVal) > 24){
+      newVal = '24';
+    } else if(Number(newVal) < 0){
+      newVal = '0';
```

Comment:
We can still add single '0'. Needs to change to '00'.

Reply:
Done

Reply:
Please note that, to fix this I have updated logic that whenever user hit backspace event, it will clear both the digit in the hours input field.